shaders/src/light_indirect.fs

```diff
@@ -101,11 +101,12 @@ vec3 diffuseIrradiance(const vec3 n) {
 // Helper function that converts the incoming Z-up world space reflection vector
 // to a Filament IBL texture lookup vector, where the top face is actually +Y.
 vec3 zUpToIblDirection(vec3 r) {
+    r = (frameUniforms.iblRotation * vec4(r, 0.0)).xyz;
```

Comment:
If `iblRotation` becomes mat3f, this would be certainly optimized out. Otherwise, the compiler is forced to carry out multiplications by zero if IEEE mode is on (as 0xNaN = NaN and it is forced to propagate it).

Reply:
4d17478